### PR TITLE
[SPARK-36040][DOCS][K8S]Add reference to kubernetes-client's version

### PR DESCRIPTION
**Pull request by yoda-mon**
### What changes were proposed in this pull request?

Add reference to kubernetes-client's version

### Why are the changes needed?

Running Spark on Kubernetes potentially has upper limitation of Kubernetes version.
I think it is better for users to notice it because Kubernetes update speed is so fast that users tends to run Spark Jobs on unsupported version.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

SKIP_API=1 bundle exec jekyll build

### CODE REVIEW

**Comment by yoda-mon** [comment]
@holdenk @dongjoon-hyun 
Could you look into this ?

**Comment by holdenk** [comment]
jenkins ok to test
LGTM I'll leave it for a bit to see if @dongjoon-hyun has any comments otherwise I'll merge it.

**Comment by holdenk** [comment]
jenkins ok to test

**Comment by yoda-mon** [comment]
Thank you for the check.

@dongjoon-hyun What is your impression of this PR ?

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ you may set up a test cluster on your local machine using
   * Be aware that the default minikube configuration is not enough for running Spark applications.
   We recommend 3 CPUs and 4g of memory to be able to start a simple Spark application with a single
   executor.
+  * Check [kubernetes-client library](https://github.com/fabric8io/kubernetes-client)'s version of your Spark environment, and it's compatibility with your Kubernetes cluster's version.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:56]: Maybe, `, and it's` -> `and its`, @yoda-mon ?

**Comment by dongjoon-hyun** [comment]
Also, cc @attilapiros

**Review by dongjoon-hyun — COMMENTED** [review]
BTW, in the PR description, the following might be misleading. Technically, it's consistent with the document of that library, but Spark 3.1.2 works with K8s 1.19.1, doesn't it? If you hit some real failures, please file a JIRA for that.
> e.g. Spark 3.1.2 uses kubernetes-client 4.12.0 and that does not support Kubernetes >= 1.19.1

And, I'd like to remove the above example sentence from the PR description.

**Comment by yoda-mon** [comment]
> BTW, in the PR description, the following might be misleading. 

You're right, and in my use case there are no problem using Spark 3.1.x with K8s 1.20. I'll remove the sentence.

(cf. https://issues.apache.org/jira/browse/SPARK-28921 might be an example but it was a little bit old problem)

**Review by attilapiros — APPROVED** [review]
LGTM

**Comment by yoda-mon** [comment]
@attilapiros Thank you for checking.
@dongjoon-hyun @holdenk  Is there any advice ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @yoda-mon and all!
Merged to master/3.2/3.1.

**Comment by dongjoon-hyun** [comment]
I added to the Apache Spark contributor group and assigned SPARK-36040 to you. Thank you again, @yoda-mon .

**Pull request closed without merging**